[jit] remove _register_* bindings from python

zdevito opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#29499 [jit] remove _register_* bindings from python**
* #29432 [jit] Simplify ScriptModule bindings.

This changes how DataParallel and trace module creation works so that
we no longer need to mutate Module class after it has been created.

The only remaining usage of register_* functions are now inside C++
tests.

Differential Revision: [D18413652](https://our.internmc.facebook.com/intern/diff/D18413652)

zdevito committed:
[jit] remove _register_* bindings from python

This changes how DataParallel and trace module creation works so that
we no longer need to mutate Module class after it has been created.

The only remaining usage of register_* functions are now inside C++
tests.

kostmo commented:
## CircleCI build failures summary
As of commit b1dc203:
* **1/1** failures introduced in this PR
* **0/1** recognized as flaky



Here are the [reasons each build failed](https://dr.pytorch.org/commit-details.html?sha1=b1dc203d8a76ceff5f3a4632abf6eb2796bc918c).

---
This comment was automatically generated by [Dr. CI](https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment).
Follow [this link to opt-out](https://dr.pytorch.org/admin/comments-opt-out.html) of these comments for your Pull Requests.


Please report bugs/suggestions on the [GitHub issue tracker](https://github.com/kostmo/circleci-failure-tracker/issues).

zdevito committed:
Update on "[jit] remove _register_* bindings from python"

[jit] remove _register_* bindings from python

This changes how DataParallel and trace module creation works so that
we no longer need to mutate Module class after it has been created.

The only remaining usage of register_* functions are now inside C++
tests.

gh-metadata: pytorch pytorch 29499 gh/zdevito/133/head
